packages/lodestar-cli/src/lodecli/cmds/dev/run.ts

```diff
@@ -0,0 +1,106 @@
+import fs, {mkdirSync} from "fs";
+import process from "process";
+import {Arguments} from "yargs";
+import deepmerge from "deepmerge";
+import {initBLS} from "@chainsafe/bls";
+import {createIBeaconConfig} from "@chainsafe/lodestar-config";
+import {params as mainnetParams} from "@chainsafe/lodestar-params/lib/presets/mainnet";
+import {params as minimalParams} from "@chainsafe/lodestar-params/lib/presets/minimal";
+import {BeaconNode} from "@chainsafe/lodestar/lib/node";
+import {createNodeJsLibp2p} from "@chainsafe/lodestar/lib/network/nodejs";
+import defaultOptions from "@chainsafe/lodestar/lib/node/options";
+import {WinstonLogger} from "@chainsafe/lodestar-utils";
+import {createEnr, createPeerId} from "../../network";
+import {initDevChain, storeSSZState} from "./utils/state";
+import rimraf from "rimraf";
+import {join} from "path";
+import {IDevOptions} from "./options";
+import {getInteropValidator} from "../validator/utils/interop/validator";
+import {ApiClientOverInstance, Validator} from "@chainsafe/lodestar-validator/lib";
+import {BeaconApi, ValidatorApi} from "@chainsafe/lodestar/lib/api/impl";
+
+/**
+ * Run a beacon node
+ */
+export async function run(options: Arguments<IDevOptions>): Promise<void> {
+  await initBLS();
+
+  options = deepmerge(defaultOptions, options) as Arguments<IDevOptions>;
+  const peerId = await createPeerId();
+  options.network.discv5.enr = await createEnr(peerId);
+
+  const config = createIBeaconConfig({
+    ...(options.chain.name === "mainnet" ? mainnetParams : minimalParams),
+  });
+  const libp2p = await createNodeJsLibp2p(peerId, options.network);
+  const logger = new WinstonLogger();
+
+  const chainDir = join(options.rootDir, "dev", "beacon");
```

Comment:
I was flip-flopping about whether it makes sense to segregate the dev on-disk config from the beacon / validator command's config. (eg: whether we should save to `$ROOT_DIR/dev/beacon` or `$ROOT_DIR/beacon`)
I think it comes down to the intention behind the dev command.

(1) If the dev command is mostly about being a convenience during development, with a focus on spinning up an ephemeral beacon node with interop-style validators, then segregating to a `dev/` directory is likely the best approach.
(2) If the dev command is mostly about being a way to run lodecli with a beacon chain and all configured validators, we likely shouldn't segregate the configs into a `dev/` directory

I think we're clearly going for 1 here, not 2, so the current strategy seems appropriate.
We might clarify the intention behind the dev command in the yargs command description or set description to false (make `dev` a hidden command)

Reply:
Well, I did it mostly to avoid overwriting witti chain/config by accident when doing some tests. Makes it easier to keep it clean also